source/extensions/filters/common/expr/context.cc

```diff
@@ -29,119 +29,455 @@ convertHeaderEntry(Protobuf::Arena& arena,
+                                if (info.sha256PeerCertificateDigest().empty()) {
+                                  return {};
+                                }
+                                return CelValue::CreateString(&info.sha256PeerCertificateDigest());
+                              }}});
+}
 
+namespace {
 absl::optional<CelValue> extractSslInfo(const Ssl::ConnectionInfo& ssl_info,
                                         absl::string_view value) {
-  if (value == TLSVersion) {
-    return CelValue::CreateString(&ssl_info.tlsVersion());
-  } else if (value == SubjectLocalCertificate) {
-    return CelValue::CreateString(&ssl_info.subjectLocalCertificate());
-  } else if (value == SubjectPeerCertificate) {
-    return CelValue::CreateString(&ssl_info.subjectPeerCertificate());
-  } else if (value == URISanLocalCertificate) {
-    if (!ssl_info.uriSanLocalCertificate().empty()) {
-      return CelValue::CreateString(&ssl_info.uriSanLocalCertificate()[0]);
-    }
-  } else if (value == URISanPeerCertificate) {
-    if (!ssl_info.uriSanPeerCertificate().empty()) {
-      return CelValue::CreateString(&ssl_info.uriSanPeerCertificate()[0]);
-    }
-  } else if (value == DNSSanLocalCertificate) {
-    if (!ssl_info.dnsSansLocalCertificate().empty()) {
-      return CelValue::CreateString(&ssl_info.dnsSansLocalCertificate()[0]);
-    }
-  } else if (value == DNSSanPeerCertificate) {
-    if (!ssl_info.dnsSansPeerCertificate().empty()) {
-      return CelValue::CreateString(&ssl_info.dnsSansPeerCertificate()[0]);
-    }
-  } else if (value == SHA256PeerCertificateDigest) {
-    if (!ssl_info.sha256PeerCertificateDigest().empty()) {
-      return CelValue::CreateString(&ssl_info.sha256PeerCertificateDigest());
-    }
+  const auto& extractors = SslExtractorsValues::get().extractors_;
+  auto it = extractors.find(value);
+  if (it != extractors.end()) {
+    return it->second(ssl_info);
   }
   return {};
 }
-
 } // namespace
 
+// Request lookup implementation
+const RequestLookupValues& RequestLookupValues::get() {
+  CONSTRUCT_ON_FIRST_USE(
+      RequestLookupValues,
+      absl::flat_hash_map<absl::string_view, CelValueExtractor>{
+          {Headers,
+           [](const RequestWrapper& wrapper) -> absl::optional<CelValue> {
+             return CelValue::CreateMap(&wrapper.headers_);
+           }},
+          {Time,
+           [](const RequestWrapper& wrapper) -> absl::optional<CelValue> {
+             return CelValue::CreateTimestamp(absl::FromChrono(wrapper.info_.startTime()));
+           }},
+          {Size,
+           [](const RequestWrapper& wrapper) -> absl::optional<CelValue> {
+             if (wrapper.headers_.value_ != nullptr &&
+                 wrapper.headers_.value_->ContentLength() != nullptr) {
+               int64_t length;
+               if (absl::SimpleAtoi(wrapper.headers_.value_->getContentLengthValue(), &length)) {
+                 return CelValue::CreateInt64(length);
+               }
+             }
```

Comment:
This fallthrough slightly changes the implementation - if the value is invalid int, it'll report `bytesReceives`. It probably makes more sense to report something in this case.